app/stores/ada/ServerConnectionStore.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import { computed } from 'mobx';
+import Store from '../base/Store';
+import type { ServerStatusErrorType } from '../../types/serverStatusErrorType';
+import environment from '../../environment';
+import { ServerStatusError } from '../../api/ada/errors';
+
+export default class ServerConnectionStore extends Store {
+  SERVER_STATUS_REFRESH_INTERVAL = 20000;
+  // TODO: set using environment.serverStatusRefreshInterval;
+
+  setup() {
+    setInterval(this._checkServerStatus, this.SERVER_STATUS_REFRESH_INTERVAL);
+  }
+
+  @computed get checkAdaServerStatus(): ServerStatusErrorType {
+    return this._checkServerStatus();
+  }
+
+  async _checkServerStatus(): ServerStatusErrorType {
```

Comment:
Should be `Promise<ServerStatusErrorType>`

Reply:
At the end it was `Promise<void>`